docs/tutorial/data-flow.md

```diff
@@ -0,0 +1,177 @@
+
+```java
+@Controller
+public class MyController {
+
+  MyComponent component;
+
+  @OnRender
+  void createSubs() {
+    component = new MyComponent(new Foo());
+    component.bar(new Data());
+    // ...
+  }
+
+  void onButtonClicked() {
+    component.foo();
+  }
+
+}
+```
+
+## From children to parents
+
+Data should usually only flow from parents to children. A controller can pass data directly to its sub-components using the `@Param` annotation, but a child should not directly access the parent instance, ensuring reusability without dependency on the parent.
+
+However, sending data from a child to the parent is sometimes necessary. For example, if a component contains a button that changes the screen color when pressed, the parent needs to be updated when the button is clicked. 
+
+### Callbacks
+An easy way of sending data from a child to a parent is creating a calback (e.g. a Consumer or Runnable) in the parent and passing it to the child.
+The callback can then be called in the child to pass data to the parent.
+
+```java
+@Component
+public class MyComponent {
+
+  @Param("colorChange")
+  Consumer<Color> color;
+
+  void onButtonClicked() {
+    Color newColor = ...;
+    color.accept(newColor);
+  }
+}
+```
+
+```java
+@Controller
+public class MyController {
+
+  @OnRender
+  void createSubs() {
+    Consumer<Color> colorChange = (color) -> System.out.println("New color is " + color);
+    MyComponent component = app.initAndRender(new MyComponent(), Map.of("colorChange", colorChange));
+    // ...
+  }
+
+}
+```
+
+The same approach can be used with Runnables instead of Consumers if only an effect needs to be triggered without passing any data.
+Instead of the `@Param` annotation, callbacks can also be set using getters, setters, or constructor parameters, though that is not very advisable.
+
+### Properties
+
+For bidirectional data flow, Properties can be used.
+Instead of defining a Runnable or Consumer directly, you can create a Property and then register listeners to it.
+
+```java
+@Component
+public class MyComponent {
+
+  @Param("colorChange")
+  ObjectProperty<Color> color;
+
+  @Param("colorBind")
+  // As this field is final, bind() will be used instead
+  final ObjectProperty<Color> otherColor = new SimpleObjectProperty<>();
+
+  void onButtonClicked() {
+    Color newColor = ...;
+    color.set(newColor);
+  }
+}
+```
+
+```java
+@Controller
+public class MyController {
+
+  Subscriber subscriber = ...;
+
+  @OnRender
+  void createSubs() {
+    ObjectProperty<Color> color = new SimpleObjectProperty(Color.WHITE);
+    ObjectProperty<Color> colorBind = new SimpleObjectProperty(Color.WHITE);
```

Comment:
E.g. here. It's possible to update colorBind and see the change in MyComponent, but not the other way around